install fail with make: *** [install] Error 1

downlad git repository
```
malonedeMacBook-Pro:delve malone$ pwd
/Users/malone/go/src/github.com/derekparker/delve
```
install follow doc 
```
malonedeMacBook-Pro:delve malone$ make install
2018/10/17 15:22:54 exit status 128
exit status 1
make: *** [install] Error 1
```
install follow another example , i create dlv-cert, but still make error
```
malonedeMacBook-Pro:delve malone$ CERT=dlv-cert make install
2018/10/17 15:41:37 exit status 128
exit status 1
make: *** [install] Error 1
```
i already read https://github.com/derekparker/delve/issues/1277,but i have no idea
 go env

```
malonedeMacBook-Pro:delve malone$ go env
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/malone/Library/Caches/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/malone/go:/Users/malone/go/src/miaopost:/Users/malone/go/src/mylib"
GOPROXY=""
GORACE=""
GOROOT="/usr/local/go"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/q3/jj46hzqj6zg6b5vdzgkhp79r0000gn/T/go-build579771309=/tmp/go-build -gno-record-gcc-switches -fno-common"
```

--- a/Documentation/installation/osx/install.md
+++ b/Documentation/installation/osx/install.md
@@ -21,6 +21,6 @@ Only do this if you have a valid reason to use the native backend.
 1. Run `xcode-select --install`
 2. On macOS 10.14 manually install the legacy include headers by running `/Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.14.pkg`
 3. Clone the repo into `$GOPATH/src/github.com/derekparker/delve`
-4. Run `make install` in that directory
+4. Run `make install` in that directory (on some versions of macOS this requires being root, the first time you run it, to install a new certificate)
 
 The makefile will take care of creating and installing a self-signed certificate automatically.

--- a/scripts/make.go
+++ b/scripts/make.go
@@ -105,22 +105,33 @@ This option can only be specified if testset is basic or a single package.`)
 	return RootCommand
 }
 
-func checkCertCmd(cmd *cobra.Command, args []string) {
+func checkCert() bool {
 	// If we're on OSX make sure the proper CERT env var is set.
 	if os.Getenv("TRAVIS") == "true" || runtime.GOOS != "darwin" || os.Getenv("CERT") != "" {
-		return
+		return true
 	}
 
 	x := exec.Command("scripts/gencert.sh")
+	x.Stdout = os.Stdout
+	x.Stderr = os.Stderr
+	x.Env = os.Environ()
 	err := x.Run()
 	if x.ProcessState != nil && !x.ProcessState.Success() {
 		fmt.Printf("An error occurred when generating and installing a new certificate\n")
-		os.Exit(1)
+		return false
 	}
 	if err != nil {
-		log.Fatal(err)
+		fmt.Printf("An error occoured when generating and installing a new certificate: %v\n", err)
+		return false
 	}
 	os.Setenv("CERT", "dlv-cert")
+	return true
+}
+
+func checkCertCmd(cmd *cobra.Command, args []string) {
+	if !checkCert() {
+		os.Exit(1)
+	}
 }
 
 func strflatten(v []interface{}) []string {
@@ -214,7 +225,9 @@ func prepareMacnative() string {
 	if !canMacnative() {
 		return ""
 	}
-	checkCertCmd(nil, nil)
+	if !checkCert() {
+		return ""
+	}
 	return "-tags=macnative"
 }
 

